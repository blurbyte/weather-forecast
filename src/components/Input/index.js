import styled from 'styled-components';
import { transparentize } from 'polished';

const Input = styled.input`
  color: ${props => props.theme.primary};
  width: 28rem;
  height: 4rem;
  border: 1px solid transparent;
  border-radius: 0;
  background-color: rgba(255, 255, 255, 0.3);
  padding: 0 1.2rem;
  font-size: 1.4rem;
  outline: 0;
  transition: all 0.3s;

  &::-ms-clear {
    display: none;
  }

  &:focus {
    width: 36rem;
    border: 1px solid #222222;
    background-color: #ffffff;
    color: #222222;
  }

  &:focus::placeholder {
    color: ${transparentize(0.5, '#222222')};
  }

  &::placeholder {
    color: ${props => transparentize(0.5, props.theme.primary)};
  }
`;

export default Input;
