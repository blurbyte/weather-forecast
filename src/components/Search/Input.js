import styled from 'styled-components';

import BaseInput from '../Input';
import Icon from './Icon';

const Input = styled(BaseInput)`
  padding-right: 4.4rem;

  & + ${Icon} {
    color: ${props => props.theme.primary};
  }

  &:focus + ${Icon} {
    color: #222;
  }
`;

export default Input;
