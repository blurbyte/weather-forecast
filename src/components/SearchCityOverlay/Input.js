import styled from 'styled-components';

const Input = styled.input.attrs({
  type: 'search',
  autoCapitalize: 'off',
  spellCheck: 'false',
  autoCorrect: 'off',
  autoComplete: 'off',
  autoFocus: true
})`
  width: 100%;
  height: 6rem;
  border: none;
  outline: none;
  border-bottom: 1px solid #222222;
  padding: 0 2rem;
  background: #ffffff;
  font-size: 1.8rem;
  color: #222222;

  &::placeholder {
    color: rgba(34, 34, 34, 0.5);
  }
`;

export default Input;
