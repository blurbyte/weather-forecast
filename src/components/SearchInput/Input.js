import styled from 'styled-components';

import BaseInput from '../Input';
import Icon from './Icon';

const Input = styled(BaseInput)`
  &:focus {
    width: 36rem;
  }

  padding-right: 4.4rem;

  & + ${Icon} {
    color: ${props => props.theme.primary};
  }

  &:focus + ${Icon} {
    color: #222;
  }
`;

export default Input;
