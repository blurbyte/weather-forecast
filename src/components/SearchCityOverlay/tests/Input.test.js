import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Input from '../Input';

test('renders correctly', () => {
  const component = renderer.create(<Input />);
  expect(component.toJSON()).toMatchSnapshot();
});
