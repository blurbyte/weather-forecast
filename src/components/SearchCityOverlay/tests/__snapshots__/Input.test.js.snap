// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 6rem;
  border: none;
  outline: none;
  border-bottom: 1px solid #222222;
  padding: 0 2rem;
  background: #ffffff;
  font-size: 1.8rem;
  color: #222222;
}

.c0::-webkit-input-placeholder {
  color: rgba(34,34,34,0.5);
}

.c0::-moz-placeholder {
  color: rgba(34,34,34,0.5);
}

.c0:-ms-input-placeholder {
  color: rgba(34,34,34,0.5);
}

.c0::placeholder {
  color: rgba(34,34,34,0.5);
}

<input
  autoCapitalize="off"
  autoComplete="off"
  autoCorrect="off"
  autoFocus={true}
  className="c0"
  spellCheck="false"
  type="search"
/>
`;
